src/node.cc

```diff
@@ -5294,47 +5289,88 @@ void RunEventLoop(const std::function<void()>& callback) {
 }
 
 
-v8::Local<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, const std::vector<v8::Local<v8::Value>> & args = {}) {
+v8::MaybeLocal<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, const std::vector<v8::Local<v8::Value>> & args) {
     return function->Call(receiver, args.size(), const_cast<v8::Local<v8::Value>*>(&args[0]));
 }
 
-v8::Local<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, std::initializer_list<v8::Local<v8::Value>> args) {
+v8::MaybeLocal<v8::Value> Call(v8::Local<v8::Object> receiver, v8::Local<v8::Function> function, std::initializer_list<v8::Local<v8::Value>> args) {
     return Call(receiver, function, std::vector<v8::Local<v8::Value>>(args));
 }
 
-// TODO: Error handling: Node.js has exceptions disabled.
-v8::Local<v8::Value> Call(v8::Local<v8::Object> object, const std::string& function_name, const std::vector<v8::Local<v8::Value>>& args) {
-  Local<v8::String> v8_function_name = v8::String::NewFromUtf8(isolate, function_name.c_str());
+v8::MaybeLocal<v8::Value> Call(v8::Local<v8::Object> object, const std::string& function_name, const std::vector<v8::Local<v8::Value>>& args) {
+  MaybeLocal<v8::String> maybe_function_name = v8::String::NewFromUtf8(isolate, function_name.c_str());
+  Local<v8::String> v8_function_name;
 
-  Local<v8::Value> value = object->Get(v8_function_name);
-  if (!value->IsFunction()) {
-    //throw new Exception(":((");
-    // TODO (js): at least return at this point
+  if (!maybe_function_name.ToLocal(&v8_function_name)) {
+    // cannot create v8 string.
+    return MaybeLocal<v8::Value>();
+  }
+
+  MaybeLocal<v8::Value> maybe_value = object->Get(v8_function_name);
+  Local<v8::Value> value;
+
+  if (!maybe_value.ToLocal(&value)) {
+    // cannot get member of object
+    return MaybeLocal<v8::Value>();
+  } else if (!value->IsFunction()) {
+    // cannot execute non-function
+    return MaybeLocal<v8::Value>();
   }
 
   return Call(object, v8::Local<v8::Function>::Cast(value), args);
 }
 
-v8::Local<v8::Value> Call(v8::Local<v8::Object> object, const std::string & function_name, std::initializer_list<v8::Local<v8::Value>> args) {
+v8::MaybeLocal<v8::Value> Call(v8::Local<v8::Object> object, const std::string & function_name, std::initializer_list<v8::Local<v8::Value>> args) {
     return Call(object, function_name, std::vector<v8::Local<v8::Value>>(args));
 }
 
-// TODO: Node.js has exceptions disabled.
-v8::Local<v8::Object> IncludeModule(const std::string& module_name) {
-  std::vector<v8::Local<v8::Value>> args = {v8::String::NewFromUtf8(isolate, module_name.c_str())};
+v8::MaybeLocal<v8::Object> IncludeModule(const std::string& module_name) {
+  MaybeLocal<v8::String> maybe_arg = v8::String::NewFromUtf8(isolate, module_name.c_str());
+  Local<v8::String> arg;
+
+  if (!maybe_arg.ToLocal(&arg)) {
+    // cannot create v8 string
+    return MaybeLocal<v8::Object>();
```

Comment:
Why don't we return `maybe_arg` here? Couldn't it carry valuable information for the application programmer?

Reply:
generally, why do we always return a new `MaybeLocal` object and not the one that failed?

Reply:
`maybe_arg` is of type `MaybeLocal<v8::String>` while the method returns `MaybeLocal<v8::Object>`. Casting one `MaybeLocal<T>` to another `MaybeLocal<V>` is only possible when casting to a `Local<T>` and then to `Local<T>`, which then can be wrapped inside a `MaybeLocal<V>`.
Long story short: Its not the way we want the code to look like. trust me